// ng2-signalr/lib/signalr.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var signalr_1 = require("./signalr");
var SignalRModule = SignalRModule_1 = (function () {
    function SignalRModule() {
    }
    SignalRModule.configure = function (configuration) {
        return {
            ngModule: SignalRModule_1,
            providers: [
                {
                    provide: signalr_1.SignalR,
                    useFactory: function (zone) { return new signalr_1.SignalR(configuration, zone); },
                    deps: [core_1.NgZone]
                }
            ],
        };
    };
    return SignalRModule;
}());
SignalRModule = SignalRModule_1 = __decorate([
    core_1.NgModule({
        providers: [{
                provide: signalr_1.SignalR,
                useValue: signalr_1.SignalR
            }]
    }),
    __metadata("design:paramtypes", [])
], SignalRModule);
exports.SignalRModule = SignalRModule;
var SignalRModule_1;
